packages/block/src/blockItemStore.ts

```diff
@@ -79,40 +79,46 @@ export class BlockItemStore<TBlock extends IBlockStub> extends StartStopService
         else this.itemsByHeight.set(blockHeight, new Set([memKey]));
         this.items.set(memKey, item);
 
-        this.batch.put(dbKey, item);
+        const dbItem = this.serialiser.serialise(item);
+
+        this.batch.put(dbKey, dbItem);
     }
 
     /**
      * Gets the item with key `itemKey` for block `blockHash`.
      * Returns `undefined` if a key is not present.
      **/
-    public getItem(blockHash: string, itemKey: string): any | undefined {
+    public getItem(blockHash: string, itemKey: string): DbObject | undefined {
         const key = `${blockHash}:${itemKey}`;
         return this.items.get(key);
     }
 
     // Type safe methods to store blocks
     public block = {
         get: (blockHash: string): TBlock | undefined =>
-            this.getItem(blockHash, BlockItemStore.KEY_BLOCK), // prettier-ignore
+            this.getItem(blockHash, BlockItemStore.KEY_BLOCK) as unknown as TBlock | undefined, // prettier-ignore
         set: (blockHeight: number, blockHash: string, block: TBlock) =>
             this.putBlockItem(blockHeight, blockHash, BlockItemStore.KEY_BLOCK, block) // prettier-ignore
     };
 
     // Type safe methods to store the "attached" boolean for each block (used in the BlockCache)
     public attached = {
         get: (blockHash: string): boolean | undefined =>
-            this.getItem(blockHash, BlockItemStore.KEY_ATTACHED), // prettier-ignore
+            this.getItem(blockHash, BlockItemStore.KEY_ATTACHED) as unknown as boolean, // prettier-ignore
         set: (blockHeight: number, blockHash: string, attached: boolean) =>
             this.putBlockItem(blockHeight, blockHash, BlockItemStore.KEY_ATTACHED, attached) // prettier-ignore
     };
 
     // Type safe methods to store the anchor state for each block, indexed by component (used in the BlockchainMachine)
     public anchorState = {
-        get: <TAnchorState>(componentName: string, blockHash: string): TAnchorState | undefined =>
-            this.getItem(blockHash, `${componentName}:${BlockItemStore.KEY_STATE}`), // prettier-ignore
-        set: (componentName: string, blockHeight: number, blockHash: string, newState: AnchorState) =>
-            this.putBlockItem(blockHeight, blockHash, `${componentName}:${BlockItemStore.KEY_STATE}`, newState)
+        get: <TAnchorState>(componentName: string, blockHash: string): TAnchorState | undefined => {
+            const rawObject = this.getItem(blockHash, `${componentName}:${BlockItemStore.KEY_STATE}`) as unknown as PlainObject;
```

Comment:
Should be DbObject not PlainObject?

Reply:
Leaving PlainObject as discussed in chat, but improved the type to `get: <TAnchorState extends PlainObjectOrSerialisable>`.